### ISSUE TITLE
Missing data sources

### ISSUE THREAD

**Issue by lwy295515016** [al3xsh/rolling-element-bearing-fault-detection#5]
Seeing your "cwru_data_loader. Py", I still don't understand how to input my own data. Could you please explain in detail how to input my own original data?

**Comment by al3xsh** [comment]
@lwy295515016,

In the training script(s), we load the data like this:

```
# load, normalise, and split the source data
data_path = args.data_path
experiment = '48k_drive_end_fault'
normalisation = 'robust-zscore'
source_data = CWRUBearingData(data_path, experiment, source, 
                              normalisation=normalisation)
source_data.change_variables(variables)
x_data, y_data, _, _ = source_data.split_data(360000, 
                                              train_fraction=1.0,
                                              window_step=64, 
                                              window_length=4096,
                                              verbose=False)

# get the train - test split
from sklearn.model_selection import train_test_split

x_train, x_test, y_train, y_test = train_test_split(x_data, y_data, 
                                                    test_size=0.1, 
                                                    random_state=42,
                                                    stratify=y_data)

# one hot encode the labels 
y_train = to_categorical(y_train)
y_test  = to_categorical(y_test)
```

The basic steps here are:

1) Normalise the data to minimise the effects of differing magnitudes (I use robust z-score normalisation). If you have lots of unique traces you might not need to do this.
2) Split the data into windows (I use a window length of 4096 samples and a step of 64 - see paper for details). Here I use the CWRU dataloader because of the complexity of the data structures in the mat files for the CWRU dataset, but if you just have vibration timeseries you can just load and window those.
3) Split into training and validation sets.

Exactly how you load your data will depend on how it is stored, but you should follow the general steps outlined above.

Regards,

Alex
